buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -1242,5 +1242,16 @@ private static boolean isUtf8(ByteBuf buf, int index, int length) {
         return true;
     }
 
+    public static int lengthOfByteOrderMark(ByteBuf buf, Charset charset) {
+        final byte[] bom = "\uFEFF".getBytes(charset);
+        int length = 0;
+        for (int i = 0; i < bom.length && i < buf.capacity(); i++) {
```

Comment:
capacity is the total amount of bytes the buffer has, shouldn't this be just the readable bytes?

Reply:
I think that's right.  Done.